tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/AspNetCoreBlockMiddlewareIntegrationEnd.cs

```diff
@@ -55,7 +55,13 @@ public static CallTargetState OnMethodBegin<TTarget>(TTarget instance)
             where TTarget : IApplicationBuilder
         {
             instance.Components.Insert(0, rd => new BlockingMiddleware(rd).Invoke);
+            if (instance.Components.Count > 1)
+            {
+                instance.Components.Insert(instance.Components.Count - 1, rd => new BlockingMiddleware(rd).Invoke);
```

Comment:
How come this middleware is needed? Is the first one not enough?

Reply:
Yeah, I don't understand the need for this. This adds the `BlockingMiddleware` at the end of the pipeline right? But that's what you do in the method below anyway. So you'll have two copies of the blocking middleware consecutively?

Reply:
My worry is that some other middleware would catch the exception and respond to it instead of ours, by placing it right after what we think is the handler, we minimize the chances of that.

On further reflection:
- we can't really be sure the last item in the chain is the handler
- another middleware handling the the exception and rendering the error page probably isn't such a bad outcome

So I removed it in b9a46c5f6